src/redux/books/Books.js

```diff
@@ -1,44 +1,35 @@
+import api from '../../api/api';
+
 const ADD_BOOK = 'ADD_BOOK';
 const REMOVE_BOOK = 'REMOVE_BOOK';
-const data = [
-  {
-    id: '1',
-    title: 'The Hunger Games',
-    author: 'Suzan Collins',
-  },
-  {
-    id: '2',
-    title: 'JavaScript',
-    author: 'Sami',
-  },
-  {
-    id: '3',
-    title: 'React Js',
-    author: 'Hassan',
-  },
-];
+const GET_BOOKS = 'GET_BOOKS';
 
-// Reducer
-const booksReducer = (state = data, action) => {
+const booksReducer = (state = [], action) => {
   switch (action.type) {
     case ADD_BOOK:
       return [...state, action.book];
     case REMOVE_BOOK:
       return [...state.filter(({ id }) => id !== action.bookId)];
+    case GET_BOOKS:
+      return action.books;
     default:
       return state;
   }
 };
 
-// Action Creators
-export const addBook = (book) => ({
-  type: ADD_BOOK,
-  book,
-});
+export const addBook = (book) => async (dispatch) => {
+  await api.addNewBook(book);
+  dispatch({ type: ADD_BOOK, book });
+};
```

Comment:
- [ ] when i add a book, I get an error requiring you to use middleware, kindly use thunk middleware to create your add book action
![Screenshot from 2022-09-21 14-26-16](https://user-images.githubusercontent.com/66097371/191516626-e9c47e66-7488-4b47-9ba1-cb284a027f74.png)

you can import the middleware like so `import { createAsyncThunk } from '@reduxjs/toolkit';`
and create your addbook like so `export const addBook = createAsyncThunk(ADD_BOOK, async (book) => {...}`

Reply:
you can also add the thunk middleware in your configureStore.js